lib/pharos/host/ubuntu/scripts/configure-cri-o.sh

```diff
@@ -29,20 +69,34 @@ else
     fi
 fi
 
-DEBIAN_FRONTEND=noninteractive apt-get install -y cri-o-$CRIO_VERSION
-systemctl enable crio
-# remove unnecessary cni plugins
-rm /etc/cni/net.d/100-crio-bridge.conf /etc/cni/net.d/200-loopback.conf || true
-systemctl start crio
+if [ ! "$(cat /etc/crio/.version)" = "$CRIO_VERSION" ]; then
+    DL_URL="https://dl.bintray.com/kontena/pharos-bin/cri-o/cri-o-v${CRIO_VERSION}-linux-amd64.tar.gz"
+    curl -sSL $DL_URL -o /tmp/cri-o.tar.gz
+    curl -sSL "${DL_URL}.asc" -o /tmp/cri-o.tar.gz.asc
+    gpg --verify /tmp/cri-o.tar.gz.asc /tmp/cri-o.tar.gz
+    tar -C / -xzf /tmp/cri-o.tar.gz
+    echo $CRIO_VERSION > /etc/crio/.version 
+fi
 
-# Install crictl binary if needed
+rm -f /etc/cni/net.d/100-crio-bridge.conf /etc/cni/net.d/200-loopback.conf || true
 
+orig_config=$(cat /etc/crio/crio.conf)
+lineinfile "^stream_address =" "stream_address = \"${CRIO_STREAM_ADDRESS}\"" "/etc/crio/crio.conf"
+lineinfile "^cgroup_manager =" "cgroup_manager = \"cgroupfs\"" "/etc/crio/crio.conf"
+lineinfile "^log_size_max =" "log_size_max = 134217728" "/etc/crio/crio.conf"
+lineinfile "^pause_image =" "pause_image = \"${IMAGE_REPO}\/pause-${CPU_ARCH}:3.1\"" "/etc/crio/crio.conf"
 
-if ! which crictl > /dev/null || [ "$(crictl -v)" != "$CRICTL_VERSION" ]; then
-    # Not installed or wrong version
-    curl -sSL https://bintray.com/kontena/pharos-bin/download_file?file_path=crictl-v${CRICTL_VERSION}-linux-${CPU_ARCH}.tar.gz -o /tmp/crictl.tar.gz
-    curl -sSL https://bintray.com/kontena/pharos-bin/download_file?file_path=crictl-v${CRICTL_VERSION}-linux-${CPU_ARCH}.tar.gz.asc -o /tmp/crictl.tar.gz.asc
-    gpg --verify /tmp/crictl.tar.gz.asc /tmp/crictl.tar.gz
-    tar xzf /tmp/crictl.tar.gz
-    install -m 755 -o root -g root crictl /usr/local/bin/crictl && rm crictl
-fi
+if ! systemctl is-active --quiet crio; then
+    systemctl enable crio
```

Comment:
Missing `systemctl daemon-reload` after writing the service unit

```
+ systemctl is-active --quiet crio
+ systemctl enable crio
Failed to execute operation: File exists
/app/lib/pharos/ssh/remote_command.rb:66:in `run!'
/app/lib/pharos/ssh/client.rb:60:in `exec!'
/app/lib/pharos/ssh/client.rb:72:in `exec_script!'
/app/lib/pharos/host/configurer.rb:70:in `exec_script'
/app/lib/pharos/host/ubuntu/ubuntu_xenial.rb:42:in `configure_container_runtime'
/app/lib/pharos/phases/configure_host.rb:22:in `call'
/app/lib/pharos/phase_manager.rb:70:in `block in apply'
/app/lib/pharos/phase_manager.rb:26:in `block (2 levels) in run_parallel'
```

```
$ systemctl status crio.service
Warning: crio.service changed on disk. Run 'systemctl daemon-reload' to reload units.
● crio.service
   Loaded: not-found (Reason: No such file or directory)
   Active: inactive (dead)
```

Reply:
Hmm... something else as well

```
root@terom-pharos-worker1:~# systemctl daemon-reload
root@terom-pharos-worker1:~# systemctl status crio.service
● crio.service - Open Container Initiative Daemon
   Loaded: loaded (/etc/systemd/system/crio.service; enabled; vendor preset: enabled)
  Drop-In: /etc/systemd/system/crio.service.d
           └─10-cgroup.conf
   Active: inactive (dead)
     Docs: https://github.com/kubernetes-incubator/cri-o
root@terom-pharos-worker1:~# systemctl enable crio.service
Failed to execute operation: File exists
```

Reply:
Is this upgrade?

Reply:
Seems like I must still have had something lying around from an earlier `cri-o-*` install, I guess?

```
root@terom-pharos-worker1:~# ls -la /etc/systemd/system/multi-user.target.wants/crio.service 
lrwxrwxrwx 1 root root 32 May 28 08:40 /etc/systemd/system/multi-user.target.wants/crio.service -> /lib/systemd/system/crio.service
```

Reply:
Does the `systemctl enable` work without the `systemctl daemon-reload`?

Reply:
If you mean the first install, yes (although not sure why). Maybe it would be safer to call `daemon-reload` also there.